test/functional/test-3p-environment.js

```diff
@@ -35,6 +35,10 @@ describe('3p environment', () => {
     });
   });
 
+  afterEach(() => {
+    testWin = null;
```

Comment:
I really dislike having to null things out inside our test environments. What if we instead nulled out our mocha suite (each `describe` block) a la https://github.com/mochajs/mocha/issues/1991#issuecomment-168379139?